app/javascript/mastodon/components/status_list.jsx

```diff
@@ -53,35 +53,58 @@ export default class StatusList extends ImmutablePureComponent {
   };
 
   handleMoveUp = (id, featured) => {
-    const elementIndex = this.getCurrentStatusIndex(id, featured) - 1;
-    this._selectChild(elementIndex, true);
+    const index = this.getCurrentStatusIndex(id, featured);
+    this._selectChild(id, index, -1);
   };
   
   handleMoveDown = (id, featured) => {
-    const elementIndex = this.getCurrentStatusIndex(id, featured) + 1;
-    this._selectChild(elementIndex, false);
+    const index = this.getCurrentStatusIndex(id, featured);
+    this._selectChild(id, index, 1);
   };
 
-  handleLoadOlder = debounce(() => {
-    const { statusIds, lastId, onLoadMore } = this.props;
-    onLoadMore(lastId || (statusIds.size > 0 ? statusIds.last() : undefined));
-  }, 300, { leading: true });
+  _selectChild = (id, index, direction) => {
+    const listContainer = this.node.node;
+    let elementContainer = listContainer.querySelector(
+      // :nth-of-type uses 1-based indexing
+      `article:nth-of-type(${index + 1 + direction})`
+    );
+    
+    if (!elementContainer) {
+      return;
+    }
+
+    // If selected container element is empty, we skip it
+    if (elementContainer.matches(':empty')) {
+      this._selectChild(id, index + direction, direction);
+      return;
+    }
 
-  _selectChild (index, align_top) {
-    const container = this.node.node;
-    // TODO: This breaks at the inline-follow-suggestions container
-    const element = container.querySelector(`article:nth-of-type(${index + 1}) .focusable`);
+    const loadMoreButton = elementContainer.querySelector('.load-more.load-gap');
+    const element = loadMoreButton ?? elementContainer.querySelector('.focusable');
 
     if (element) {
-      if (align_top && container.scrollTop > element.offsetTop) {
-        element.scrollIntoView(true);
-      } else if (!align_top && container.scrollTop + container.clientHeight < element.offsetTop + element.offsetHeight) {
-        element.scrollIntoView(false);
+      const elementRect = element.getBoundingClientRect();
+
+      const columnHeaderHeight = 60;
```

Comment:
this feels fragile- while the value is set in the CSS below, that doesn't account for things like custom themes. instead i'd suggest calculating the height using JS.

Reply:
I changed it so the value is now read from CSS so there's a single source of truth. Custom themes will have to set the custom property if needed.